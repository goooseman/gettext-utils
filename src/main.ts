// tslint:disable-next-line
import "module-alias/register";
import exportStrings from "./exportStrings";
import importStrings from "./importStrings";

export { exportStrings, importStrings };
