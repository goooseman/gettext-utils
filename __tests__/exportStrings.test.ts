import { exportStrings } from "@src/main";
import * as fse from "fs-extra";
import * as os from "os";
import * as path from "path";

beforeAll(() => {
  global.Date.prototype.toString = () => "Thu Jan 01 2018 00:00:00";
});

const templateName = "template.pot";

it("Should export correct strings from .tsx files", async () => {
  const tmpPath = await getTmpPath();
  await exportStrings(
    tmpPath,
    "__fixtures__/react-project/src/**/{*.ts,*.tsx,*.js,*.jsx}",
  );
  expect(readFileFromTmp(tmpPath, templateName)).toMatchSnapshot(templateName);
});

it("Should overwrite template.pot", async () => {
  const tmpPath = await getTmpPath();
  const filePath = path.join(tmpPath, templateName);
  await fse.createFile(filePath);
  await exportStrings(
    tmpPath,
    "__fixtures__/react-project/src/**/{*.ts,*.tsx,*.js,*.jsx}",
  );
  expect(readFileFromTmp(tmpPath, templateName)).toMatchSnapshot(templateName);
});

it("Should update .po files with new translations", async () => {
  const tmpPath = await getTmpPath();
  await fse.copy("__fixtures__/po", tmpPath);
  await exportStrings(
    tmpPath,
    "__fixtures__/react-project/src/**/{*.ts,*.tsx,*.js,*.jsx}",
  );
  const locales = ["fr.po", "he.po"];

  for (const locale of locales) {
    expect(readFileFromTmp(tmpPath, locale)).toMatchSnapshot(locale);
  }
});

const readFileFromTmp = (tmpPath: string, filePath: string) => {
  return fse.readFileSync(path.join(tmpPath, filePath)).toString();
};

const getTmpPath = async () => {
  const exportFolder = path.join(os.tmpdir(), `lioness-utils`);
  await fse.remove(exportFolder);
  return exportFolder;
};
