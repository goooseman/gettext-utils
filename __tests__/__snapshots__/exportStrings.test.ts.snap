// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should export correct strings from .jsx and .tsx files: template.pot 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Project-Id-Version: lioness-utils 0.0.1\\\\n\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"POT-Creation-Date: Thu Jan 01 2018 00:00:00\\\\n\\"
\\"Content-Transfer-Encoding: 8bit\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n != 1);\\\\n\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:19
msgctxt \\"lion.sound\\"
msgid \\"Rrrrr\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:23
msgctxt \\"lion.females\\"
msgid \\"one female lion\\"
msgid_plural \\"{{ count }} female lions\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:22
msgctxt \\"lion.title\\"
msgid \\"Lion\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:23
msgctxt \\"lion.children\\"
msgid \\"Lion has one child\\"
msgid_plural \\"Lion has {{ count }} children\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Should overwrite template.pot: template.pot 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Project-Id-Version: lioness-utils 0.0.1\\\\n\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"POT-Creation-Date: Thu Jan 01 2018 00:00:00\\\\n\\"
\\"Content-Transfer-Encoding: 8bit\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n != 1);\\\\n\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:19
msgctxt \\"lion.sound\\"
msgid \\"Rrrrr\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:23
msgctxt \\"lion.females\\"
msgid \\"one female lion\\"
msgid_plural \\"{{ count }} female lions\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:22
msgctxt \\"lion.title\\"
msgid \\"Lion\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:23
msgctxt \\"lion.children\\"
msgid \\"Lion has one child\\"
msgid_plural \\"Lion has {{ count }} children\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Should update .po files with new translations: fr.po 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"POT-Creation-Date: Thu Jan 01 2018 00:00:00\\\\n\\"
\\"Content-Transfer-Encoding: 8bit\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n > 1);\\\\n\\"
\\"Project-Id-Version: lioness-utils 0.0.1\\\\n\\"
\\"PO-Revision-Date: \\\\n\\"
\\"Language-Team: \\\\n\\"
\\"MIME-Version: 1.0\\\\n\\"
\\"X-Generator: Poedit 2.2.1\\\\n\\"
\\"Last-Translator: \\\\n\\"
\\"Language: fr\\\\n\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:19
msgctxt \\"lion.sound\\"
msgid \\"Rrrrr\\"
msgstr \\"Oooo\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:23
msgctxt \\"lion.females\\"
msgid \\"one female lion\\"
msgid_plural \\"{{ count }} female lions\\"
msgstr[0] \\"lionne\\"
msgstr[1] \\"{{ count }} lionnes\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:22
msgctxt \\"lion.title\\"
msgid \\"Lion\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:23
msgctxt \\"lion.children\\"
msgid \\"Lion has one child\\"
msgid_plural \\"Lion has {{ count }} children\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Should update .po files with new translations: he.po 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"POT-Creation-Date: Thu Jan 01 2018 00:00:00\\\\n\\"
\\"Content-Transfer-Encoding: 8bit\\\\n\\"
\\"Plural-Forms: nplurals=4; plural=(n==1 ? 0 : n==2 ? 1 : n>10 && n%10==0 ? 2 \\"
\\": 3);\\\\n\\"
\\"Project-Id-Version: lioness-utils 0.0.1\\\\n\\"
\\"PO-Revision-Date: \\\\n\\"
\\"Language-Team: \\\\n\\"
\\"MIME-Version: 1.0\\\\n\\"
\\"X-Generator: Poedit 2.2.1\\\\n\\"
\\"Last-Translator: \\\\n\\"
\\"Language: he\\\\n\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:19
msgctxt \\"lion.sound\\"
msgid \\"Rrrrr\\"
msgstr \\"רררר\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:23
msgctxt \\"lion.females\\"
msgid \\"one female lion\\"
msgid_plural \\"{{ count }} female lions\\"
msgstr[0] \\"לביאה\\"
msgstr[1] \\"שתי לביאה\\"
msgstr[2] \\"‫{{count}} לביאות\\"
msgstr[3] \\"‏‫{{count}} לביא\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:22
msgctxt \\"lion.title\\"
msgid \\"Lion\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:23
msgctxt \\"lion.children\\"
msgid \\"Lion has one child\\"
msgid_plural \\"Lion has {{ count }} children\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;

exports[`Should update .po files with new translations: ru.po 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=utf-8\\\\n\\"
\\"POT-Creation-Date: Thu Jan 01 2018 00:00:00\\\\n\\"
\\"Content-Transfer-Encoding: 8bit\\\\n\\"
\\"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && \\"
\\"n%10<=4 && (n%100<12 || n%100>14) ? 1 : 2);\\\\n\\"
\\"Project-Id-Version: lioness-utils 0.0.1\\\\n\\"
\\"PO-Revision-Date: \\\\n\\"
\\"Language-Team: \\\\n\\"
\\"MIME-Version: 1.0\\\\n\\"
\\"X-Generator: Poedit 2.2.1\\\\n\\"
\\"Last-Translator: \\\\n\\"
\\"Language: ru\\\\n\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:19
msgctxt \\"lion.sound\\"
msgid \\"Rrrrr\\"
msgstr \\"Ррррр\\"

#: __fixtures__/react-project/src/components/LionessHocComponent.tsx:23
msgctxt \\"lion.females\\"
msgid \\"one female lion\\"
msgid_plural \\"{{ count }} female lions\\"
msgstr[0] \\"одна львица\\"
msgstr[1] \\"{{ сount }} львицы\\"
msgstr[2] \\"{{ count }} львиц\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:22
msgctxt \\"lion.title\\"
msgid \\"Lion\\"
msgstr \\"\\"

#: __fixtures__/react-project/src/components/LionessTComponent.jsx:23
msgctxt \\"lion.children\\"
msgid \\"Lion has one child\\"
msgid_plural \\"Lion has {{ count }} children\\"
msgstr[0] \\"\\"
msgstr[1] \\"\\""
`;
